apollo-router/src/axum_factory/handlers.rs

```diff
@@ -166,7 +168,7 @@ where
                                 .into_response()
                         }
                         Some(response) => {
-                            if response.has_next.unwrap_or(false) {
+                            if response.has_next.unwrap_or(false) || is_multipart {
```

Comment:
there should be a special case if there is only one response but we must send it in multipart. Here on lines 179 to 183 we create the first part

```
--graphql
content-type: application/json

{ ... }

\r\n--graphql\r\n
```

But since there will not be any other response, we should send 

```
--graphql
content-type: application/json

{ ... }

\r\n--graphql--\r\n
```

note the extra `--` after `graphql`, marking the end of the last part, otherwise added on line 196